### PR TITLE
fct_products_summary correction

### PR DESCRIPTION
**Pull request by AnasErad**

### CODE REVIEW

File: dbt/models/facts/fct_products_summary.sql [hunk 1]
```diff
@@ -1,40 +1,119 @@
...
+        -- product info 
+        i.value['product']['id']                                                                  AS PRODUCT_ID,
+        null                                                                                      AS PRODUCT_NAME,
+        t.ID                                                                                      AS ORDER_ID,
+        'TAPPAYMENTS'                                                                             AS REVENUE_SOURCE,
+        t.ERAD_REVENUE                                                                            AS REVENUE,
+        i.value['quantity']                                                                       AS QUANTITY
+
+        FROM {{ref('stg_revenue_tappayments')}} as t,
+        LATERAL FLATTEN(input => t.ITEMS ) i
```
Comment (leonDataNut) [inline dbt/models/facts/fct_products_summary.sql:75]: NICEEEEE

File: dbt/models/facts/fct_products_summary.sql [hunk 2]
```diff
@@ -1,40 +1,122 @@
...
+    cte_prefix='order') }},
 
-SELECT 
+{{build_fivetran_ctes_from_one_schema_table(
+    schema_base_name = 'SHOPIFY', 
+    table_name = 'ORDER_LINE', 
+    columns = ['ID',	'ORDER_ID',	'PRODUCT_ID','QUANTITY'],
+    cte_prefix='orderline') }},
```
Comment (leonDataNut) [inline dbt/models/facts/fct_products_summary.sql:24]: These should follow the same flow as the revenue summary table. Each of these CTEs should be their own stage table view.

Once the stage tables are done there should be 2 intermediate tables created , one for sales and one for payments. From there we create a product intermediate table that goes through decision tree to decide if we should use the sales or payments data. See documentation here: https://www.notion.so/eradco/Dashboard-and-data-transformations-9d1083b6d42b4d85b63d04326ec6d968

File: dbt/models/facts/fct_products_summary.sql [hunk 3]
```diff
@@ -1,40 +1,122 @@
...
+        'TITLE',PRODUCT_UNION.TITLE,
+        'HANDLE',PRODUCT_UNION.HANDLE,
+        'PRODUCT_TYPE',PRODUCT_UNION.PRODUCT_TYPE,
+        'VENDOR',PRODUCT_UNION.VENDOR,
+        'CREATED_AT',PRODUCT_UNION.CREATED_AT,
+        'UPDATED_AT',PRODUCT_UNION.UPDATED_AT,
+        'PUBLISHED_AT',PRODUCT_UNION.PUBLISHED_AT,
+        'PUBLISHED_SCOPE',PRODUCT_UNION.PUBLISHED_SCOPE,
+        'TEMPLATE_SUFFIX',PRODUCT_UNION.TEMPLATE_SUFFIX,
+        'STATUS', PRODUCT_UNION.STATUS)                                                           AS PRODUCT,
```
Comment (leonDataNut) [inline dbt/models/facts/fct_products_summary.sql:47]: I like the idea of an object here, but have 1 change. Can you make product name a dedicated column and rename this object column to DETAILS

File: dbt/models/facts/fct_products_summary.sql [hunk 4]
```diff
@@ -1,40 +1,122 @@
...
-m.PRODUCT                                  AS PRODUCT,
-TO_VARCHAR(s.ORDER_ID)                     AS ORDER_ID,
-'SHOPIFY'                                  AS REVENUE_SOURCE,
-s.REVENUE_USD                              AS REVENUE,
-s.QUANTITY                                 AS QUANTITY
 
-FROM {{ref('stg_products_shopify')}} as s */
+        ORDERLINE_UNION.ORDER_ID                                                                  AS ORDER_ID,
+        'SHOPIFY'                                                                                 AS REVENUE_SOURCE,
+        {{ convert_to_erad_currency('CURRENCY' ,'ORDER_UNION.TOTAL_PRICE_USD') }}                 AS REVENUE_USD,
```
Comment (leonDataNut) [inline dbt/models/facts/fct_products_summary.sql:53]: This should be accounted for in the stage table

File: dbt/models/staging/revenue/stg_productsSummary_shopify.sql [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        p.HANDLE,
+        p.PRODUCT_TYPE,
+        p.VENDOR,
+        p.CREATED_AT,
+        p.UPDATED_AT,
+        p.PUBLISHED_AT,
+        p.PUBLISHED_SCOPE,
+        p.TEMPLATE_SUFFIX,
+        p.STATUS,
+        p."_FIVETRAN_DELETED"
```
Comment (leonDataNut) [inline dbt/models/staging/revenue/stg_productsSummary_shopify.sql:17]: Can you remove the quotes?

File: dbt/models/staging/revenue/stg_productsSummary_shopify.sql [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        o.CUSTOMER_ID,
+        o.TOTAL_PRICE_USD,
+        o.CREATED_AT
+    FROM {{ref('stg_order_shopify')}} as o
+
+                ),
+
+ORDERLINE_UNION AS (
+    SELECT 
+        ol.ID,
```
Comment (leonDataNut) [inline dbt/models/staging/revenue/stg_productsSummary_shopify.sql:34]: Can you explain why we have to use to datasets for shopify? Not saying this is wrong just want to understand.

File: dbt/models/staging/revenue/stg_productsSummary_shopify.sql [hunk 7]
```diff
@@ -0,0 +1,77 @@
+WITH
```
Comment (leonDataNut) [inline dbt/models/staging/revenue/stg_productsSummary_shopify.sql:1]: Should the name of this model be stg_products_shopify?

File: dbt/models/staging/revenue/stg_productsSummary_shopify.sql (hunk 5 above)
Comment (AnasErad) [reply dbt/models/staging/revenue/stg_productsSummary_shopify.sql:17]: done

File: dbt/models/staging/revenue/stg_productsSummary_shopify.sql (hunk 6 above)
Comment (AnasErad) [reply dbt/models/staging/revenue/stg_productsSummary_shopify.sql:34]: didn't understand your question

**Pull request merged**